inc/classes/ServiceProvider/class-third-party-subscribers.php

```diff
@@ -61,5 +62,7 @@ public function register() {
 			->withArgument( $this->getContainer()->get( 'options' ) );
 		$this->getContainer()->share( 'optimus_webp_subscriber', 'WP_Rocket\Subscriber\Third_Party\Plugins\Images\Webp\Optimus_Subscriber' );
 		$this->getContainer()->share( 'bigcommerce_subscriber', 'WP_Rocket\Subscriber\Third_Party\Plugins\Ecommerce\BigCommerce_Subscriber' );
+		$this->getContainer()->share( 'amp_subscriber', 'WP_Rocket\ThirdParty\Plugins\Optimization\AMP' )
+			->withArgument( $this->getContainer()->get( 'options' ) );
```

Comment:
Each of these subscribers is using a separate instance of options. That's less performant.

Since we're touching this file, let's make it more performant. Create one instance and then use that variable in each of these:

```php
$options = $this->getContainer()->get( 'options' );
```

Reply:
done in https://github.com/wp-media/wp-rocket/pull/2459/commits/218a81f02ceb1fce826e9c5c11519975f4174a47